src/main/java/hudson/plugins/git/GitStatus.java

```diff
@@ -313,6 +326,45 @@ public void writeBody(PrintWriter w) {
         }
 
         /**
+         * Check if there is a ParameterValue with the same name
+         *
+         */
+        private boolean parameterValueExist (List<ParameterValue> parameterValues, String name) {
+            for(ParameterValue parameterValue : parameterValues) {
+                if (parameterValue.getName().equals(name)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /**
+         * Get the default parameters values from a job
+         *
+         */
+        private List<ParameterValue> getDefaultParametersValues(Job job) {
+            ParametersDefinitionProperty paramDefProp = (ParametersDefinitionProperty) job.getProperty(ParametersDefinitionProperty.class);
+            ArrayList<ParameterValue> defValues = new ArrayList<ParameterValue>();
```

Comment:
How often are we triggering builds? If triggering builds is a hot operation, and most builds have more than 10 parameters and computing the max size is cheap, then giving an initial size is a good plan.

I think we are missing quite a few of these conditions here